File: Tyche.iOS/User/Sources/User/Login/UI/LoginCredentialModel.swift
import Foundation

struct LoginCredentialModel : Equatable {
    var username: String
    var password: String
}

extension LoginCredentialModel {
    
    func isValid() -> Bool {
        return Username.isValid(value: self.username)
        && Password.isValid(value: self.password)
    }
    
    func toInput() -> LoginInput {
        return LoginInput(
            username: Username(value: self.username)!,
            password: Password(value: self.password)!
        )
    }
}

File: Tyche.Android/user/src/main/java/com/felipearpa/tyche/user/login/ui/LoginCredentialModel.kt
package com.felipearpa.tyche.user.login.ui

import com.felipearpa.tyche.user.UserProfile
import com.felipearpa.tyche.user.type.Password
import com.felipearpa.tyche.user.type.Username

data class LoginCredentialModel(
    val username: String,
    val password: String
)

fun LoginCredentialModel.hasErrors(): Boolean {
    return !(Username.isValid(this.username) && Password.isValid(this.password))
}